homeassistant/components/discovergy/sensor.py

```diff
@@ -165,33 +172,27 @@ async def async_setup_entry(
 
     entities: list[DiscovergySensor] = []
     for meter in meters:
-        sensors = None
-        if meter.measurement_type == "ELECTRICITY":
-            sensors = ELECTRICITY_SENSORS
-        elif meter.measurement_type == "GAS":
-            sensors = GAS_SENSORS
-
+        sensors: tuple[DiscovergySensorEntityDescription, ...] = ()
         coordinator: DiscovergyUpdateCoordinator = data.coordinators[meter.meter_id]
 
-        if sensors is not None:
-            for description in sensors:
-                # check if this meter has this data, then add this sensor
-                for key in {description.key, *description.alternative_keys}:
-                    if key in coordinator.data.values:
-                        entities.append(
-                            DiscovergySensor(key, description, meter, coordinator)
-                        )
+        # select sensor descriptions based on meter type and combine with additional sensors
+        if meter.measurement_type == "ELECTRICITY":
+            sensors = ELECTRICITY_SENSORS + ADDITIONAL_SENSORS
+        elif meter.measurement_type == "GAS":
+            sensors = GAS_SENSORS + ADDITIONAL_SENSORS
 
-        for description in ADDITIONAL_SENSORS:
-            entities.append(
-                DiscovergySensor(description.key, description, meter, coordinator)
-            )
+        entities.extend(
+            DiscovergySensor(value_key, description, meter, coordinator)
+            for description in sensors
+            for value_key in {description.key, *description.alternative_keys}
+            if description.value_fn(coordinator.data, value_key, description.scale)
+        )
 
     async_add_entities(entities, False)
```

Comment:
```suggestion
    async_add_entities(entities)
```

Reply:
Wouldn't trigger this a second refresh, I already call `coordinator.async_config_entry_first_refresh()` in `__init__.py`?

Reply:
Nvm. It is per default `False`

Reply:
The default is False, no?

Reply:
Yep, the default is `False`.